akka-stream/src/main/scala/akka/stream/impl/fusing/AggregateWithTimeBoundary.scala

```diff
@@ -49,10 +43,8 @@ class AggregateWithBoundary[T, Agg, Emit](
     new TimerGraphStageLogic(shape) with InHandler with OutHandler {
 
       private[this] var aggregated: Agg = null.asInstanceOf[Agg]
-
-      private val buffer = mutable.Queue[Emit]()
-
-      private def bufferFull: Boolean = buffer.size >= bufferSize
+      private[this] var hasEmitted: Boolean = false
```

Comment:
could if it has started an aggregate but not emitted be inferred by whether the `aggregated` is null or not (also, was it impossible to use `OptionVal` for `aggregated` instead of explicit `null` for some reason?)

Reply:
I think this was the tricky part I didn't do it right. This code is already removed. `hasEmitted` only worked for initial pull, need the internal buffer for the general case.
used `OptionVal` before, but the syntax of using `OptionVal` is more verbose than using `null`. It's also more transparent this way. If `OptionVal` can support the 'map', `foreach` and `exists' syntactic sugar that would be better.